automated install/basic-install.sh

```diff
@@ -360,7 +360,11 @@ if is_command apt-get ; then
     PIHOLE_DEPS=(cron curl iputils-ping lsof netcat psmisc sudo unzip wget idn2 sqlite3 libcap2-bin dns-root-data libcap2)
     # Packages required for the Web admin interface (stored as an array)
     # It's useful to separate this from Pi-hole, since the two repos are also setup separately
-    PIHOLE_WEB_DEPS=(lighttpd "${phpVer}-common" "${phpVer}-cgi" "${phpVer}-${phpSqlite}" "${phpVer}-xml" "${phpVer}-json" "${phpVer}-intl")
+    PIHOLE_WEB_DEPS=(lighttpd "${phpVer}-common" "${phpVer}-cgi" "${phpVer}-${phpSqlite}" "${phpVer}-xml" "${phpVer}-intl")
+    # Prior to PHP8.0, JSON functionality is provided as dedicated module, required by Pi-hole AdminLTE: https://www.php.net/manual/json.installation.php
+    if [[ "${phpInsNewer}" != true || "${phpInsMajor}" -lt 8 ]]; then
```

Comment:
So, my undestanding here is that `phpInsNewer` will be true only if major version is  greater than, or equal to, 7.

What you're effectively saying here is "add `php-json` if php major version is less than 7", which doesn't seem right.

See [code above](https://github.com/pi-hole/pi-hole/pull/4135/files#diff-595630a29a855f6d667a84ca0662042e826bf3ec56322ef61d4a6ef149147d23R325-R327):

```
if [ "${phpInsMajor}" -ge 7 ]; then
    phpInsNewer=true
fi
```

Reply:
`phpInsNewer` is only set within the `if is_command php` block and `[ "${phpInsMajor}" -ge 7 ]` is then true on every supported Debian and Ubuntu version. On unsupported distros that ship PHP5 natively, the installer fails anyway since a longer while as `php5-xml` does not exist. Hence `phpInsNewer` is practically true whenever PHP is installed already.

So the code says: "Add phpX.Y-json when no PHP is installed yet or one below 8.0"
This is intended since:
- If no PHP is installed yet, any of PHP7.0 - PHP7.4 would be installed on every supported Debian/Ubuntu, which means that the JSON module is a separate package which needs to be installed. In theory, one could add https://deb.sury.org/ to make PHP8.0 the newly installed PHP version, but I doubt that anyone would manually add this repo without actually installing PHP from it before running the Pi-hole installer. However, that case wasn't supported before either.
- If PHP is installed already, but it's PHP8.0 from https://deb.sury.org/ (linked case in the forum) or a custom compiled PHP8.0, `"${phpInsMajor}" -lt 8` is false, so the JSON package is not installed by intention, as `php8.0-json` does not exist.

If someone managed to install PHP5 on a supported distro (which all natively serve PHP7.0 at least), thanks to https://github.com/pi-hole/pi-hole/pull/4135/files#diff-595630a29a855f6d667a84ca0662042e826bf3ec56322ef61d4a6ef149147d23R331 the regular distro meta packages, i.e. PHP7.0 - PHP7.4 would be installed additionally.

The only way this could be an issue is when someone uses https://deb.sury.org/ but installed PHP5. Then the meta packages would pull PHP8.0 but this code would try to install `php-json`, which only exists as PHP7.X meta package, hence the user most likely ends up with three PHP installs:
- PHP5 manually installed before on a distro where PHP7 is natively available and via a 3rd party repo that provides PHP8.
- PHP7 pulled with `php-json` meta package.
- PHP8 pulled with all other `php-*` meta packages.
- But: It would be a functional setup without installer errors 😅.

To rule that case out (and support the case that https://deb.sury.org/ is installed but no PHP from it), the retrieved PHP version would need to be checked before installing it without the CLI available, which is kind of nasty via PHP cache and implies other possible failures. I think the chance that someone has PHP5 installed the above way is much lower than the chance that someone has PHP8 installed, now that it has officially been released. So I see the current implementation as best compromise between supporting PHP8 in most cases without introducing possible regressions.

Reply:
Right, OK, so by that logic...

User already has `php7.2` installed before running the Pi-hole installer.

`phpInsNewer` gets set to true, `phpVer` gets set to `php7.3`.... and `php7.3-json` never gets installed...

Reply:
> `phpInsNewer` gets set to true, `phpVer` gets set to `php7.3`.... and `php7.3-json` never gets installed...

`phpInsMajor < 8`, so it does get installed, not the `||` (or) operator:
```
if [[ "${phpInsNewer}" != true || "${phpInsMajor}" -lt 8 ]]; then
```

I have to admin the negations and operators are not to intuitive. It would have been better understandable when removing the package from the list conditionally, rather than adding it. But from the code point of view adding to an array is so much easier than removing from an array. See the CentOS pre-7 section where this is done for `php-json`, as the module was compiled into the core package on earlier CentOS versions.

Reply:
Oh piss! I was  reading `||` as `&&` for no other reason than I misread it.

Carry on...!